ReactNativeClient/lib/joplin-renderer/MdToHtml/rules/copy_to_clipboard.js

```diff
@@ -0,0 +1,108 @@
+const fs = require('fs');
+const path = require('path');
+const filePath = path.join(__dirname, '../copyFunction.js');
+const btnCode = fs.readFileSync(filePath).toString();
+
+
+const copyCodeStyle = `
+.hljs {
+	position: relative;
+}
+.copy-wrapper {
+  display: none;
+  position: absolute;
+  top: 0;
+  right:0;
+}
+.copy-wrapper button {
+	padding: 8px 8px;
+	background: #fff;
+	margin: 5px 8px;
+	border: none;
+	cursor: pointer
+}
+.copy-wrapper button i {
+	font-size: 0.98rem;
+}
+
+  .hljs:hover .copy-wrapper{
+	  display: inline !important;
+  } 
+
+.u-mdic-copy-notify {
+	font-size: 0.75rem;
+	color: #cccccc;
+	display: none;
+	background: #fff;
+	border-radius: 10px;
+	padding: 5px 10px;
+}
+`;
+
+
+
+function generateUuid() {
+	return `${+Date.now()}-${parseInt(Math.random() * 100000)}`;
+}
+
+const LAST_TAG = '</pre>';
+
+const enhance = (render, options = {}) => (...args) => {
+	/* args = [tokens, idx, options, env, slf] */
+	const {
+		failText = 'copy fail', // copy-fail text
+		successText = 'copied!', // copy-success text
+		successTextDelay = 2000, // successText show time [ms]
+	} = options;
+	const [tokens = {}, idx = 0] = args;
+	// get content
+	const cont = tokens[idx].content;
+	const uuid = `j-notify-${generateUuid()}`;
+	const buttonBuilder = [
+		'<div class="copy-wrapper">',
+		`<div class="u-mdic-copy-notify" id="${uuid}">${successText}</div>`,
+		'<button ',
+		'class="u-mdic-copy-btn j-mdic-copy-btn" ',
+		`data-mdic-content="${cont}" `,
+		`data-mdic-notify-id="${uuid}" `,
+		`data-mdic-notify-delay="${successTextDelay}" `,
+		`data-mdic-copy-fail-text="${failText}" `,
+		`onclick="${btnCode}"><i class="fa fa-copy"></i></button>`,
+		'</div>',
+	];
+	const copyTag = buttonBuilder.join('');
+
+	const originResult = render.apply(this, args);
+	const newResult = originResult.replace(LAST_TAG, `${copyTag}${LAST_TAG}`);
```

Comment:
We shouldn't assume that the code block ends with a </PRE>. What if it's changed to a DIV later, or what if there are multiple </PRE> inside the block? (which I think can happen) Is there a more reliable way to insert this content without making assumption about how it ends?

Reply:
i'll look into this

Reply:
Good day @laurent22 , please can you suggest a way I could insert the
button into the <pre> element. This is my only challenge now